backends/hip-ref/ceed-hip-ref-vector.c

```diff
@@ -14,89 +14,68 @@
 #include "ceed-hip-ref.h"
 
 //------------------------------------------------------------------------------
-// Sync host to device
-//------------------------------------------------------------------------------
-static inline int CeedVectorSyncH2D_Hip(const CeedVector vec) {
-  int ierr;
-  Ceed ceed;
-  ierr = CeedVectorGetCeed(vec, &ceed); CeedChkBackend(ierr);
-  CeedVector_Hip *impl;
-  ierr = CeedVectorGetData(vec, &impl); CeedChkBackend(ierr);
-
-  CeedSize length;
-  ierr = CeedVectorGetLength(vec, &length); CeedChkBackend(ierr);
-  size_t bytes = length * sizeof(CeedScalar);
-
-  if (!impl->h_array)
+// Get size of the scalar type
+// TODO: move to interface level for all backends?
+//------------------------------------------------------------------------------
+static inline int CeedScalarTypeGetSize_Hip(Ceed ceed, CeedScalarType prec,
+    size_t *size) {
+  switch(prec) {
+  case CEED_SCALAR_FP32:
+    *size = sizeof(float);
+    break;
+  case CEED_SCALAR_FP64:
+    *size = sizeof(double);
+    break;
+  default:
```

Comment:
I put this in the backend for now while testing with hip-ref, but is this something that could be useful/necessary to have at an interface level for all backends, rather than each `ref` backend duplicating it?